### DISCUSSION TITLE
YouTube recorded support. Https vs http websites.

### DISCUSSION THREAD

**Discussion by siliconhippy** [streamlink/streamlink#5454] [Q&A]
### Checklist

- [X] This is a feature request and not a different kind of issue
- [X] [I have read the contribution guidelines](https://github.com/streamlink/streamlink/blob/master/CONTRIBUTING.md#contributing-to-streamlink)
- [X] [I have checked the list of open and recently closed plugin requests](https://github.com/streamlink/streamlink/issues?q=is%3Aissue+label%3A%22feature+request%22)

### Description

I am following up on an earlier issue, which was closed due to relation to a fork ...

First I did research on this site and found out that proxy servers, both http(s) and socks are supported ( at least in upcoming version). 

Don't know about web sockets support, but maybe this is irrelevant here.

https://streamlink.github.io/latest/cli/proxy.html

1. Is YouTube recorded video streaming supported ?

The problem I heard was that YouTube recorded videos ( as opposed to live tv channels?) are split streamed into separate audio and video streams which then have to be assembled by ffmpeg on various clients like the popular kodi.tv and Jellyfin media server.

Is there going to some future fix, if it is even possible?

3. Can the streamlink native package correctly stream https based websites like mjunoon.tv through http proxy like squid?

A. Is https proxy necessary? Or is there a workaround? 
B. How about for through a socks5 with remote DNS proxy ( e.g., Dante + squid+ Adguard Home) that is SSL encrypted between external proxy client ( like the Kodi or Jellyfin media server) and proxy server?

Hope I am violating any guidelines !

**Comment by bastimeyer** [comment]
> 1\. Is YouTube recorded video streaming supported ?

The kinds of streams which the YouTube plugin supports are HLS streams and adaptive HTTP streams that get muxed by ffmpeg automatically. The adaptive streams however are bandwidth-limited because they are lacking a signature token (YouTube restriction which still needs to be resolved). YouTube's DASH streams are unsupported.

- https://streamlink.github.io/cli/tutorial.html
- https://streamlink.github.io/plugins.html#youtube

> 3\. Can the streamlink native package correctly stream https based websites like mjunoon.tv through http proxy like squid?

https://streamlink.github.io/cli/proxy.html

It doesn't matter for the chosen proxy whether the HTTP request that gets tunneled is HTTP or HTTPS.

**Reply by siliconhippy** [reply]
Ok thanks 🙏 
So I will assume that YouTube recorded channels are not really supported yet by streamlink.
Also that the mjunoon.tv http proxy block issue is not streamlink related.

**Reply by siliconhippy** [reply]
Also if you can reply 😄

I should be able to use this same pipe method to use streamlink installed on same Linux machine as Kodi.tv to stream channels via Kodi Simple IPTV PVR?

See TVH link / Building m3u lists:
https://cgomesu.com/blog/Tvhlink/#tvhlink

( This TVH Link, btw, has nothing to do with TVlink in AlexElec.tv !)
